core/templates/pages/skill-editor-page/editor-tab/skill-prerequisite-skills-editor/skill-prerequisite-skills-editor.component.ts

```diff
@@ -13,123 +13,153 @@
+  skillIdToSummaryMap: object;
+  prerequisiteSkillsAreShown: boolean;
+  skill: Skill;
+  allAvailableSkills: SkillSummary[] = [];
+  @Output() updateView = new EventEmitter();
+
+  constructor(
+    private skillUpdateService: SkillUpdateService,
+    private skillEditorStateService: SkillEditorStateService,
+    private alertsService: AlertsService,
+    private topicsAndSkillsDashboardBackendApiService:
+      TopicsAndSkillsDashboardBackendApiService,
+    private windowDimensionsService: WindowDimensionsService,
+    private ngbModal: NgbModal
+  ) {}
+
+  removeSkillId(skillId: string): void {
+    this.skillUpdateService.deletePrerequisiteSkill(this.skill, skillId);
+    this.updateView.emit();
+  }
+
+  getSkillEditorUrl(skillId: string): string {
+    return '/skill_editor/' + skillId;
+  }
+
+  addSkill(): void {
+    // This contains the summaries of skill in the same topic as
+    // the current skill as the initial entries followed by the others.
+    const skillsInSameTopicCount =
+    this.groupedSkillSummaries.current.length;
+    const sortedSkillSummaries = this.groupedSkillSummaries.current.concat(
+      this.groupedSkillSummaries.others);
+    const allowSkillsFromOtherTopics = true;
+
+    const modalRef: NgbModalRef = this.ngbModal.open(
+      SelectSkillModalComponent, {
+        backdrop: 'static',
+        windowClass: 'skill-select-modal',
+        size: 'xl'
+      });
+
+    modalRef.componentInstance.skillSummaries = sortedSkillSummaries;
+    modalRef.componentInstance.skillsInSameTopicCount = (
+      skillsInSameTopicCount);
+    modalRef.componentInstance.categorizedSkills = this.categorizedSkills;
+    modalRef.componentInstance.allowSkillsFromOtherTopics = (
+      allowSkillsFromOtherTopics);
+    modalRef.componentInstance.untriagedSkillSummaries = (
+      this.untriagedSkillSummaries);
+
+    const whenResolved = (summary): void => {
+      let skillId = summary.id;
+      if (skillId === this.skill.getId()) {
+        this.alertsService.addInfoMessage(
+          'A skill cannot be a prerequisite of itself', 5000);
+        return;
+      }
+      for (let idx in this.skill.getPrerequisiteSkillIds()) {
+        if (this.skill.getPrerequisiteSkillIds()[idx] === skillId) {
+          this.alertsService.addInfoMessage(
+            'Given skill is already a prerequisite skill', 5000);
+          return;
         }
-      };
-    }]
+      }
+      this.skillUpdateService.addPrerequisiteSkill(this.skill, skillId);
+      this.updateView.emit();
+    };
+
+    modalRef.result.then(function(summary) {
+      whenResolved(summary);
+    }, function() {
+      // Note to developers:
+      // This callback is triggered when the Cancel button is clicked.
+      // No further action is needed.
+    });
+  }
+
+  togglePrerequisiteSkills(): void {
+    if (this.windowDimensionsService.isWindowNarrow()) {
+      this.prerequisiteSkillsAreShown =
+        !this.prerequisiteSkillsAreShown;
```

Comment:
Just a pattern I noticed, it seems like the parenthesis are being removed, please make sure to use extra parenthesis if a statement breaks into multiple lines? (see [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#javascript))

Reply:
done, I will take care of this from next point onward